internal/controllers/clusterextension_controller.go

```diff
@@ -470,16 +472,31 @@ func (d *DefaultInstalledBundleGetter) GetInstalledBundle(ctx context.Context, e
 		return nil, err
 	}
 
-	release, err := cl.Get(ext.GetName())
+	rel, err := cl.Get(ext.GetName())
 	if err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
 		return nil, err
 	}
-	if release == nil {
+	if rel == nil {
 		return nil, nil
 	}
 
+	switch {
+	case rel.Info.Status == release.StatusUnknown:
+		return nil, fmt.Errorf("installation status is unknown")
+	case rel.Info.Status == release.StatusDeployed:
+	case rel.Info.Status == release.StatusUninstalled:
+	case rel.Info.Status == release.StatusSuperseded:
+	case rel.Info.Status == release.StatusFailed:
```

Comment:
If the release is failed, do we want to allow it to be overridden?

Reply:
Based on my understanding of https://github.com/operator-framework/operator-controller/issues/1296 I would assume we would not want to do this.

Reply:
It's in a terminal state; and if an upgrade is the only possible way to fix it, then we need to allow it to happen. Otherwise, the installation will remain in a failed state and be unable to be replaced without manual intervention (i.e. removal).